cl_dll/rainbow.cpp

```diff
@@ -0,0 +1,216 @@
+#include <algorithm>
+#include "hud.h"
+#include "cl_util.h"
+#include "rainbow.h"
+
+#undef min
+#undef max
+
+void CRainbow::Init()
+{
+    // Register cvars
+    m_pCvarRainbow = CVAR_CREATE("hud_rainbow", "0", FCVAR_ARCHIVE);
+    m_pCvarRainbowS = CVAR_CREATE("hud_rainbow_sat", "100", FCVAR_ARCHIVE);
+    m_pCvarRainbowV = CVAR_CREATE("hud_rainbow_val", "100", FCVAR_ARCHIVE);
+    m_pCvarRainbowSpeed = CVAR_CREATE("hud_rainbow_speed", "40", FCVAR_ARCHIVE);
+    m_pCvarRainbowXPhase = CVAR_CREATE("hud_rainbow_xphase", "0.4", FCVAR_ARCHIVE);
+    m_pCvarRainbowYPhase = CVAR_CREATE("hud_rainbow_yphase", "0.7", FCVAR_ARCHIVE);
+
+    // Save engine functions
+    m_pfnSPR_Set = gEngfuncs.pfnSPR_Set;
```

Comment:
Is it in any way possible for this to be called while functions are hooked? E.g. `_restart`. Because then it'll break.

Reply:
No, it's not. Engine unloads the client when shutting down. And if unloading fails, then `Initialize`, which initializes `gEngfuncs`, is always called before `Init`.